### PR TITLE
BUG: Resampler.ohlc with empty data

### LINKED ISSUE #42902: BUG: groupby.ohlc returns Series rather than DataFrame if input is empty
- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of pandas. (tested 1.3.1)
- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
# This returns a pd.Series with 0 items
pd.DataFrame.from_dict({
    'timestamp': pd.DatetimeIndex([]),
    'symbol': np.array([], dtype=object),
    'price': np.array([], dtype='f4'),
}).set_index('timestamp').groupby('symbol').resample('D', closed='right')['price'].ohlc()

# This returns a pd.DataFrame with columns open, high, low, close and a single row (index=[(ABC, 2021-01-01)])
pd.DataFrame.from_dict({
    'timestamp': pd.DatetimeIndex(['2021-01-01']),
    'symbol': np.array(['ABC'], dtype=object),
    'price': np.array([100], dtype='f4'),
}).set_index('timestamp').groupby('symbol').resample('D', closed='right')['price'].ohlc()
```

#### Problem description

It is surprising that `groupby.ohlc` returns different data types depending on whether or not there are zero rows.

#### Expected Output

I think it would be more sensible for the zero row case to return a `DataFrame` with columns open/high/low/close, but with zero rows. Returning a `Series` as a special case doesn't make sense.

#### Output of ``pd.show_versions()``

(This failed with an error about llvmlite.dll in the environment with pandas 1.3.1 installed, so this output is actually from a slightly earlier install of Pandas which still exhibits the bug.)

<details>
INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.8.10.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18363
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 7, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United Kingdom.1252

pandas           : 1.2.4
numpy            : 1.20.2
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.1.1
setuptools       : 52.0.0.post20210125
Cython           : 0.29.23
pytest           : 6.2.3
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.0
IPython          : 7.22.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 0.9.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.4
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : 0.18.0
xlrd             : 2.0.1
xlwt             : None
numba            : 0.53.0
</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #42902 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/resample/test_base.py [hunk 1]
```diff
@@ -199,7 +200,15 @@ def test_resample_empty_dataframe(empty_frame_dti, freq, resample_method):
 
     rs = df.resample(freq, group_keys=False)
     result = getattr(rs, resample_method)()
-    if resample_method != "size":
+    if resample_method == "ohlc":
+        # TODO: no tests with len(df.columns) > 0
```
Comment (rhshadrach) [inline pandas/tests/resample/test_base.py:204]: Can these be added here?

Comment (jbrockmendel) [reply pandas/tests/resample/test_base.py:204]: im falling behind on things.  mind if i punt on this?

Comment (rhshadrach) [reply pandas/tests/resample/test_base.py:204]: Yep - sounds good.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Just a whatsnew conflict.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**